src/utils/c_utils/src/string_c_utils.c

```diff
@@ -464,10 +464,13 @@ int ADUC_AllocAndStrCopyN(char** dest, const char* src, size_t srcLen)
         goto done;
     }
 
+    // Uses ADUC_Safe_StrCopyN to prevent buffer overflows and ensure null termination.
     num_copied = ADUC_Safe_StrCopyN(
         tmpDest, src, (srcLen + 1) * sizeof(char) /* destByteLen */, srcLen /* numSrcCharsToCopy */);
+
     if (num_copied < srcLen)
     {
+        // it was truncated!
```

Comment:
silently truncating a string seems really bad to me.

char path[8 + 1];
// what happens if we access "/this/is" ?!?
ADUC_AllocAndStrCopyN("/this/is/a/full/path", path, sizeof(path))

Reply:
Good suggestion. I've removed implicit truncation and return an error instead (made return type an int where 0 means success.

Reply:
caller is supposed to use return value to check for truncation, but I think you're right that just returning an error here would be better.  fixed.